### DISCUSSION TITLE
Error using Spawn: "FAILspawn not supported without process manager"

### DISCUSSION THREAD

**Discussion by fenfold** [mpi4py/mpi4py#151] [Q&A] [answered]
I'm getting the following error using `MPI.Spawn()`: "FAILspawn not supported without process manager".
I tried the [example](https://mpi4py.readthedocs.io/en/stable/tutorial.html#dynamic-process-management) in the tutorial but also e.g. `MPI.COMM_SELF.Spawn("dir")` doesn't work.

My system is Windows 10 x64, Python 3.9.5, with MS-MPI. This used to work for me before. I'm not sure what changed. I also tried earlier versions of mpi4py, but this didn't help.

**Comment by dalcinl** [comment, marked as answer]
I'm not a Windows expert. However, you are probably using MS MPI, which is a MPICH derivative. The error you got most likely means that you need to execute your code using `mpiexec -n <nprocs> python your_script.py`, where `<nprocs>` is the number of MPI processes you want to run, as discussed in the docs [[link](https://mpi4py.readthedocs.io/en/stable/tutorial.html#running-python-scripts-with-mpi)].

**Comment by fenfold** [comment]
Thanks. Somehow I was under the impression that calling the MPI runner was handled under the hood in mpi4py.
